src/core/repoIdRegistry.test.js

```diff
@@ -53,13 +61,69 @@ describe("core/repoIdRegistry", () => {
         {repoId: makeRepoId("zoo", "zod")},
         {repoId: makeRepoId("foo", "bar")},
       ];
-      expect(addRepoId(registry, {repoId: makeRepoId("zoo", "zod")})).toEqual([
+      expect(addEntry(registry, {repoId: makeRepoId("zoo", "zod")})).toEqual([
         {repoId: makeRepoId("foo", "bar")},
         {repoId: makeRepoId("zoo", "zod")},
       ]);
     });
   });
+
+  describe("getEntry", () => {
+    it("returns the matching entry by RepoId", () => {
+      const entry = {repoId: makeRepoId("zoo", "zod")};
+      const registry = addEntry(emptyRegistry(), entry);
+      expect(getEntry(registry, entry.repoId)).toBe(entry);
+    });
+    it("returns undefined if there is no matching entry", () => {
+      expect(getEntry(emptyRegistry(), makeRepoId("foo", "bar"))).toBe(
+        undefined
+      );
+    });
+  });
+
   it("empty registry is empty", () => {
     expect(emptyRegistry()).toEqual([]);
   });
+
+  describe("{get,write}Registry", () => {
+    it("getRegistry returns empty registry if nothing present", () => {
+      const dirname = tmp.dirSync().name;
+      expect(getRegistry(dirname)).toEqual(emptyRegistry());
+    });
+
+    it("writeRegistry writes a repoIdRegistry to the directory", () => {
+      const dirname = tmp.dirSync().name;
+      const repoId = makeRepoId("foo", "bar");
+      const entry = {repoId};
+      const registry = addEntry(emptyRegistry(), entry);
+      const registryFile = path.join(dirname, REPO_ID_REGISTRY_FILE);
+
+      expect(fs.existsSync(registryFile)).toBe(false);
+      writeRegistry(registry, dirname);
+      expect(fs.existsSync(registryFile)).toBe(true);
+
+      const contents = fs.readFileSync(registryFile);
+      const registryJSON = JSON.parse(contents.toString());
+      expect(toJSON(registry)).toEqual(registryJSON);
+    });
+
+    it("getRegistry returns the registry written by writeRegistry", () => {
+      const dirname = tmp.dirSync().name;
+      const repoId = makeRepoId("foo", "bar");
+      const entry = {repoId};
+      const registry = addEntry(emptyRegistry(), entry);
+      writeRegistry(registry, dirname);
+      expect(getRegistry(dirname)).toEqual(registry);
+    });
+
+    it("writeRegistry overwrites any existing registry", () => {
+      const dirname = tmp.dirSync().name;
+      const repoId = makeRepoId("foo", "bar");
+      const entry = {repoId};
+      const registry = addEntry(emptyRegistry(), entry);
+      writeRegistry(registry, dirname);
+      writeRegistry(emptyRegistry(), dirname);
+      expect(getRegistry(dirname)).toEqual(emptyRegistry());
+    });
+  });
```

Comment:
You could save four lines 🎉by capturing `repoId` `entry` and `registry`: 
```javascript 
  describe("{get,write}Registry", () => {
    const repoId = () => makeRepoId("foo", "bar");
    const entry = () => ({repoId: repoId()});
    const registry = () => addEntry(emptyRegistry(), entry());

    it("getRegistry returns empty registry if nothing present", () => {
      const dirname = tmp.dirSync().name;
      expect(getRegistry(dirname)).toEqual(emptyRegistry());
    });

    it("writeRegistry writes a repoIdRegistry to the directory", () => {
      const dirname = tmp.dirSync().name;
      const registryFile = path.join(dirname, REPO_ID_REGISTRY_FILE);

      expect(fs.existsSync(registryFile)).toBe(false);
      writeRegistry(registry(), dirname);
      expect(fs.existsSync(registryFile)).toBe(true);

      const contents = fs.readFileSync(registryFile);
      const registryJSON = JSON.parse(contents.toString());
      expect(toJSON(registry())).toEqual(registryJSON);
    });

    it("getRegistry returns the registry written by writeRegistry", () => {
      const dirname = tmp.dirSync().name;
      writeRegistry(registry(), dirname);
      expect(getRegistry(dirname)).toEqual(registry());
    });

    it("writeRegistry overwrites any existing registry", () => {
      const dirname = tmp.dirSync().name;
      writeRegistry(registry(), dirname);
      writeRegistry(emptyRegistry(), dirname);
      expect(getRegistry(dirname)).toEqual(emptyRegistry());
    });
  });
});
```

The benefit might be improved legibility when you remove the repetition of setting up those variables in each block.

Reply:
Done, thanks for the suggestoin.